src/shared_library.cpp

```diff
@@ -76,6 +76,7 @@ std::string SharedLibrary::get_library_path()
 std::string get_platform_library_name(std::string library_name)
 {
   char library_name_platform[1024];
+  memset(library_name_platform, 0, 1024);
```

Comment:
Using `std::array` here would be more idiomatic C++:

```c++
std::array<char, 1024> library_name_platform{};
```

Either way, you could use "value initialization" to make sure it is zeroed without using `memset`:

```c++
char library_name_platform[1024]{};
```

See: https://stackoverflow.com/a/18295840/671658

If we had C++17, then you could have used `std::string library_name_platform(1024, '\0');`, but the non-const `std::basic_string::data` method wasn't added until C++17...